common/src/main/java/com/wynntils/core/features/FeatureManager.java

```diff
@@ -291,58 +281,101 @@ private void initializeFeature(Feature feature) {
 
-        // flag as event listener
-        if (MethodUtils.getMethodsWithAnnotation(featureClass, SubscribeEvent.class).length > 0) {
-            feature.setupEventListener();
-        }
-
-        // set feature category
+        // Set feature category
         ConfigCategory configCategory = feature.getClass().getAnnotation(ConfigCategory.class);
         Category category = configCategory != null ? configCategory.value() : Category.UNCATEGORIZED;
         feature.setCategory(category);
 
-        // register key binds
-        for (Field f : FieldUtils.getFieldsWithAnnotation(featureClass, RegisterKeyBind.class)) {
-            if (!f.getType().equals(KeyBind.class)) continue;
+        // Register key binds
+        Managers.KeyBind.discoverKeyBinds(feature);
 
-            try {
-                KeyBind keyBind = (KeyBind) FieldUtils.readField(f, feature, true);
-                feature.setupKeyHolder(f.getName(), keyBind);
-            } catch (Exception e) {
-                WynntilsMod.error("Failed to register KeyBind " + f.getName() + " in " + featureClass.getName(), e);
-            }
-        }
-
-        // determine if feature should be enabled & set default enabled value for user features
+        // Determine if feature should be enabled & set default enabled value for user features
         boolean startDisabled = featureClass.isAnnotationPresent(StartDisabled.class);
         if (feature instanceof UserFeature userFeature) {
             userFeature.userEnabled = !startDisabled;
         }
 
-        // init overlays before ConfigManager
-        feature.initOverlays();
-
-        // register & load configs
-        // this has to be done after the userEnabled handling above, so the default value registers properly
-        Managers.Config.registerFeature(feature);
-        Managers.Storage.registerStorageable(feature);
+        Managers.Overlay.discoverOverlays(feature);
+        Managers.Overlay.discoverOverlayGroups(feature);
 
-        feature.initOverlayGroups();
-
-        // initialize & enable
-        feature.init();
+        // Assert that the feature name is properly translated
+        assert !feature.getTranslatedName().startsWith("feature.wynntils.");
 
         if (feature instanceof UserFeature userFeature) {
             if (!userFeature.userEnabled) return; // not enabled by user
 
-            userFeature.enable();
+            enableFeature(feature);
         } else if (!startDisabled) {
-            feature.enable();
+            enableFeature(feature);
         }
     }
 
+    public void enableFeature(Feature feature) {
+        if (!FEATURES.containsKey(feature)) {
+            throw new IllegalArgumentException("Tried to enable an unregistered feature: " + feature);
+        }
+
+        FeatureState state = FEATURES.get(feature);
+
+        if (state != FeatureState.DISABLED && state != FeatureState.CRASHED) return;
+
+        WynntilsMod.postEvent(new FeatureStateChangeEvent.Enabled(feature));
```

Comment:
I think this would, after all, be better as a callback into the feature, `feature.onEnable()` than an event. The critical point here is if anyone is listening to the event for any other reason than to handle its own feature. I'll check the code and come back here.

Reply:
Ok, back after checking. It is used in two places. One is indeed checking for "this". The other one did not, but should have. 

Conclusion: This should not be an event. The old `onEnable()` and `onDisable()` callbacks on the Feature itself is better suited for this purpose. 

Also, there is no need for an `onCrash()`. Features should not crash, so coding for special handling if it happens is kind of weird. It should be enough that `onDisable()` is called in case of a crash.